### PR TITLE
Added %timer1% to %timer16% variables to rules

### PR DESCRIPTION
**Pull request by alexasf**
## Description:

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [ ] The pull request is done against the latest development branch
  - [ ] Only relevant files were touched
  - [ ] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with Tasmota core ESP32 V.2.0.2.1
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/xdrv_10_rules.ino [hunk 1]
```diff
@@ -471,6 +472,15 @@ bool RulesRuleMatch(uint8_t rule_set, String &event, String &rule, bool stop_all
...
     }
 #endif  // USE_TIMERS and USE_SUNRISE
+#if defined(USE_TIMERS)
+    for (uint32_t i = 0; i < MAX_TIMERS; i++) {
+      snprintf_P(stemp, sizeof(stemp), PSTR("%%TIMER%d%%"), i +1);
+      if (rule_param.startsWith(stemp)) {
+        rule_param = String(Settings->timer[i].time);
+        break;
+      }
+    }
```
Comment (barbudor) [inline tasmota/xdrv_10_rules.ino:482]: Please optimize this
Comparing to 16 strings is non sense
Search first for `%timer` and only if found, look for the number and the ending

Comment (alexasf) [reply tasmota/xdrv_10_rules.ino:482]: Thank you
I optimize.

**Comment by ascillato** [comment]
What is the use-case of this?

Please, can you give an example where you need this?

**Comment by alexasf** [comment]
I use this to turn a relay on and off after a power failure. If this happened during the time interval of the timer, then the state of the output will remain unchanged.
In the rules there are only variables for setting and rising. I didn't find another way.
This is how I use it:
Rule1
  ON System#Init DO Power1 1 ENDON
  ON Time#Initialized DO Backlog var3 0;var5 0;event checktime=%time% ENDON
  ON event#checktime>%timer1% DO var3 1 ENDON
  ON event#checktime>=%timer2%  DO var3 0 ENDON
  ON event#checktime>%timer3% DO var5 1 ENDON
  ON event#checktime>=%timer4%  DO var5 0 ENDON
  ON event#checktime DO Backlog Power3 %var3%;Power5 %var5% ENDON

File: tasmota/xdrv_10_rules.ino (hunk 1 above)
Comment (alexasf) [reply tasmota/xdrv_10_rules.ino:482]: optimize mohno like this: 
    if (rule_param.startsWith(F("%TIMER"))) {
      int index = rule_param.substring(5).toInt();
      if (index > 0 && index <= MAX_TIMERS) {
         snprintf_P(stemp, sizeof(stemp), PSTR("%%TIMER%d%%"), index);
         if (rule_param.startsWith(stemp))
           rule_param = String(Settings->timer[index-1].time);
      }
    }
this option will run faster and take more code.

**Comment by ascillato** [comment]
Hi, 

Thanks. The use case is good. Just for reference, another workaround to this common use-case has been deployed using scripts ( https://github.com/arendst/Tasmota/discussions/14525 ). Anyway, it is good to have your solution too.

**Comment by alexasf** [comment]
Another use case that I applied.
Rule2
  ON Power1#State=0 DO RuleTimer1 %timer16%*60 ENDON
  ON Rules#Timer=1 DO Power1 on ENDON
  ON Power3#State=0 DO RuleTimer2 %timer15%*60 ENDON
  ON Rules#Timer=2 DO Power3 on ENDON

**Comment by rapi3** [comment]
there is a problem described here:
https://github.com/arendst/Tasmota/discussions/15205

also it would be nice if the value of timer will be stored calculated as defined by user: time/sunset/sunrise +/-value and not only if it is a fixed value

**Pull request merged**